src/components/admin/ArticleController.jsx

```diff
@@ -363,6 +384,9 @@ export default class ArticleController extends FalcorController {
           autoScrollBodyContent
           onRequestClose={this.handleDialogClose}
         >
+          <div>
```

Comment:
I'm just gonna make this comment on behalf of all the places you added this.

First of all you don't need to wrap it in an extra div.

And second of all, as you can maybe imagine it would be hell if we ever wanted to update how the `savingScreen` that we would have to update the same code in 5 different places, so let's pull it out into it's own component. Make a new file in this folder called `LoadingOverlay`, you can just make it a [stateless functional component](https://reactjs.org/docs/components-and-props.html#functional-and-class-components), and then import it in the other files and put it in as you did here.

In the end these 3 lines should look like this:
```js
{ this.state.saving ? <LoadingOverlay /> : null }
```

I think I like it better like this now that I think about it as it otherwise looks like at a quick glance that the saving screen is always there, though it of course functions.